app/models/financial_aid/my_awards.rb

```diff
@@ -209,18 +209,33 @@ def sub_items(award_type, item)
           alertDetails: alert_details(item['item_type']),
           termDetails: disbursements(item['item_type']),
           awardMessage: item['award_message'],
+          authFailedMessage: auth_failed_message(item['item_type']),
           itemTypeDescr: item['title']
         }
       else
         {
           alertDetails: alert_details(item['item_type']),
           remainingAmount: (item['left_col_amt'] - item['right_col_amt']).to_f,
           awardMessage: item['award_message'],
+          authFailedMessage: auth_failed_message(item['item_type']),
           itemTypeDescr: item['title']
-        } 
+        }
       end
     end
 
+    def auth_failed_message_exists?(item_type)
+      exists ||= EdoOracle::FinancialAid::Queries.get_auth_failed_message(@uid, my_aid_year, item_type)
```

Comment:
@marcpechaitis - You're not storing the results of this query in an instance variable. You'd want to declare it as `@exists` for it to persist... however the challenge here is that this is a dynamic function, where the result depends on the `my_aid_year` and `item_type` arguments.

You might want to instead do this:
```ruby
def auth_failed_message_exists?(item_type)
  # initialize instance variable with empty hash if it doesn't exist
  @exists ||= {}

  # initialize my aid year object if it doesn't exist
  @exists[my_aid_year] ||= {}

  # put results of aid year + item_type specific result into multidimensional hash
  @exists[my_aid_year][item_type] ||= !!EdoOracle::FinancialAid::Queries.get_auth_failed_message(@uid, my_aid_year, item_type)

  # last statement in method gets automatically returned
  @exists[my_aid_year][item_type]
end
```

Actually, I just checked and it looks like the 'my_aid_year' value is specific to the instance (just like the UID), so maybe you don't need to nest a hash within another hash... but you get the principle right?

This will result in the results of those queries are cached within the instance object while it persists in memory.

Reply:
Thanks for the explanation and insight into this @redconfetti